CareKitStore/CareKitStore/StoreCoordinator/OCKPersistentStoreCoordinator+Outcomes.swift

```diff
@@ -45,7 +45,7 @@ extension OCKStoreCoordinator {
         }
 
         let sortDescriptor = NSSortDescriptor(
-            keyPath: \OCKAnyOutcome.id,
+            keyPath: \OCKCDOutcome.id,
```

Comment:
I have some thoughts here...

It looks like the public sort descriptor for `OCKOutcomeQuery` was removed in 3.0. My guess is the code above is here to maintain some consistency when the same query is made when the exact same data is in the database?

I think there is at least one case (maybe two) to add the public SortDescriptor back. In #553 `OCKOutcome` was turned into a versioned entity, but querying for `OCKOutcome` doesn't necessarily follow the [comments](https://github.com/carekit-apple/CareKit/issues/357#issuecomment-584334257) and [documentation](https://github.com/carekit-apple/CareKit?tab=readme-ov-file#carekitstore-) which can make since due to the nature of an outcome (may want to know the result for a particular day as oppose to the latest version). Still, there are times where a developer will want results by `effectiveDate`, which all current queries to versioned objects currently allow, except for `OCKOutcome`. Because of this, I've added extensions in my helper framework to help handle `OCKOutcome` results https://github.com/netreconlab/CareKitEssentials/pull/12

If you agree, we can add the public SortDescriptor back, but include `effectiveDate` like: https://github.com/carekit-apple/CareKit/blob/7416cce107875e58d0dbfd54afb16e007ef79033/CareKitStore/CareKitStore/Structs/Queries/OCKTaskQuery.swift#L36-L64

Another case can be added to the SortDescriptor for `taskOccurrenceIndex`, which could be useful. We can then add the case from sorting by `id` (the above code), but mark that one as internal as I can imagine this could over time. We can prepend the sort by `id` to all queries to replicate the current behavior of the above block of code.

Let me know what you think and I can create a follow-up PR after a decision is made on this one.